### PR TITLE
Port esp2866 driver to new WiFiInterface API

### PR DESCRIPTION
**Pull request by bulislaw**
@0xc0170 Don't merge before mbed/feature_wifi branch is merged to master.

### CODE REVIEW

**Comment by 0xc0170** [comment]
cc @geky please review

File: ESP8266/ESP8266.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ bool ESP8266::disconnect(void)
 const char *ESP8266::getIPAddress(void)
 {
     if (!(_parser.send("AT+CIFSR")
-        && _parser.recv("+CIFSR:STAIP,\"%[^\"]\"", _ip_buffer)
+        && _parser.recv("+CIFSR:STAIP,\"%15[^\"]\"", _ip_buffer)
```
Comment (geky) [inline ESP8266/ESP8266.cpp]: +1 for this :+1:

**Comment by 0xc0170** [comment]
@bulislaw We shall update this on top of latest master (rc1 in this case) to reflect changes we did to the wifi

**Comment by c1728p9** [comment]
I rebased to master and solved merge conflicts.

**Pull request merged**